.github/workflows/build.yml

```diff
@@ -8,7 +8,8 @@ name: Build
 # This is only allowing pushes on the moonbeam repo for pull requests.
 ####### DO NOT CHANGE THIS !! #######
 on:
-  push:
+  pull_request:
+    types: [opened, edited, unassigned, synchronize]
```

Comment:
Why limit to those types ?

Reply:
Because if the build workflow is re-triggered for all types of changes, it will often be re-triggered when it's not necessary, like when we modify labels for example. See the complete list of types here: https://docs.github.com/en/actions/reference/events-that-trigger-workflows#pull_request